### PR TITLE
Simplify the animation logic within Scene

### PR DESCRIPTION
**Pull request by eulertour**
Major changes include:
* Removed the `background` argument from `CairoRenderer.update_frame()`. It was only used once, to return a static image which was later passed back to the `CairoRenderer` and composed with an image that moved as the animation progressed. I replaced its functionality with `CairoRenderer.save_static_mobject_data()`, which saves the image to the renderer instead.

* Added `Scene.get_moving_and_static_mobjects()` to simplify the logic prior to beginning the animation loop in which static and moving mobjects are separated.

* Removed `Scene.begin_animations()`, `Scene.finish_animations()`, `Scene.clean_up_animations()`, `Scene.progress_through_animations()`, and `Scene.update_animation_to_time()`. They were each only used once and were not very complicated, so I moved their contents into `Scene.play_internal()`.

* Removed [these lines](https://github.com/ManimCommunity/manim/blob/master/manim/scene/scene.py#L764-L768) from the logic that was previously in `Scene.finish_animations()`. AFAICT hey don't do anything.

* Removed several instance variables from Scene. `Scene.mobjects_from_last_animation` was only used once, `Scene.last_t` was only used in an antipattern way to pass data to a function.

Depends on #148

### CODE REVIEW

**Comment by leotrs** [comment]
Do you think this can wait until after the release? Given that we don't have enough tests for all the CLI flags, I'm a bit hesitant to touch the rendering logic right before a release.

**Comment by eulertour** [comment]
Sure, but we can still get the review done until then.

Depends on #148

**Comment by eulertour** [comment]
I feel like the dep bot is never working 😕

File: manim/scene/scene.py
Comment (huguesdevimeux) [inline manim/scene/scene.py:766]: Just a quick question about these removed lines. 
Suppose that animation n is skipped, but not animation n+1, and animation n+1 uses some stuff that were supposedly updated by animation n, in any way. Will this work without these lines? The mobject that was supposed to be updated in animation n wouldn't have been.

Comment (eulertour) [reply manim/scene/scene.py:766]: Manim handles animation skipping by setting the time progression used in the animation loop to one which holds a single value, the length of the animations being played.

https://github.com/ManimCommunity/manim/blob/3b6519a58678eb84bb31f550a0a4734df58bded3/manim/scene/scene.py#L562-L566

so the relevant mobjects will have already been updated by the time they reach this point.

**Review by behackl — APPROVED** [review]
Apart from a missing docstring, this looks good to me, thank you for your effort with this cleanup!

I don't necessarily agree with you overriding the "2 reviewers"-rule and merging this without any approvals. To be fair, you asked several times and I could have indicated that I planned looking at this. At the same time I don't think it would have halted any further progress if this PR would have been open for a little longer.

**Comment by behackl** [comment]
`git bisect` says that merging this introduced a problem with `Mobject.become`: The example

```
class FadeInAndOut(Scene):
    def construct(self):
        square = Square(color=BLUE).shift(2 * UP)
        annotation = Text("Fade In", height=0.8)
        self.add(annotation)
        self.play(FadeIn(square))

        annotation.become(Text("Fade Out", height=0.8))
        self.add(annotation)
        self.play(FadeOut(square))
```
from our documentation renders as

FadeInAndOut

Same thing if the call to `become` is replaced by playing a `Transform` animation; the old text is simply kept on screen.

**Comment by eulertour** [comment]
Where in our documentation is this example?

**Comment by behackl** [comment]
> Where in our documentation is this example?

It's not yet included, it will be added with #576.

**Pull request merged**
